### DISCUSSION TITLE
Frontend for the Pages

### DISCUSSION THREAD

**Discussion by Shivansh-yadav13** [Shivansh-yadav13/SQLite-UI#6] [Ideas]
- Need Frontend design for the Pages (figma)
- Should we use a CSS framework?

**Comment by Shivansh-yadav13** [comment]
@hamees-sayed @wliao008 @VanTekken @MagnaibayarMN How should we proceed for the UI part? need your advice.

**Reply by MagnaibayarMN** [reply]
I would recommend **TailwindCSS** for CSS framework.

**Comment by wliao008** [comment]
We definitely should use a css framework to make things consistent and easier, altho I'm no front end guy, I think the dashboard layout should work, with a sidebar panel listing all tables etc. Something like this?

<img width="224" alt="Screen Shot 2022-10-13 at 10 26 49 AM" src="https://user-images.githubusercontent.com/803646/195624323-e14e4c73-0df3-4d7c-8a76-e0f46c4091b0.png">

**Reply by Shivansh-yadav13** [reply]
I was also thinking for a similar UI :+1:

**Reply by VanTekken** [reply]
If we use a JavaScript framework like React, I'd recommend styled-components and framer-motion!

**Comment by VanTekken** [comment]
We could decouple the front and back end and use a JavaScript framework for the front end. Though, the JavaScript framework isn't absolutely necessary. To decouple the front and back end, we would need to make the Go app act as an API which the front end can fetch resources from using JavaScript 🚀

**Reply by Shivansh-yadav13** [reply]
@VanTekken I was also planning do the same but then I was thinking to keep it simple with html and css and with adding a JavaScript framework might make the project a little complex with lots of files 😅
but we can use a CSS framework.

**Reply by VanTekken** [reply]
Ok, should we make the front end separate and call the app using JavaScript or continue using templates?

**Reply by wliao008** [reply]
I like the idea of starting simple and not introduce too much complexity from the get go (but I'm biased as I'm mostly backend). But I can see as this gets used more, separating the front end then would make more sense to improve requests & user experience.

**Reply by VanTekken** [reply]
I think separating the front and back end would simplify things in the long run, but it would mean making big changes to the current code base. So let's stick with the template approach for now.
